modules/candidate_parameters/jsx/DiagnosisEvolution.js

```diff
@@ -0,0 +1,172 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Loader from 'Loader';
+import StaticDataTable from 'jsx/StaticDataTable';
+
+/**
+ * Candidate date of birth component
+ */
+class DiagnosisEvolution extends Component {
+  /**
+   * @constructor
+   * @param {object} props - React Component properties
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      error: false,
+      isLoaded: false,
+    };
+
+    this.fetchData = this.fetchData.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.formattedDiagnosisEvolution =
+    this.formattedDiagnosisEvolution.bind(this);
+    this.renderLatestDiagnosis =
+    this.renderLatestDiagnosis.bind(this);
+  }
+
+  /**
+   * Called by React when the component has been rendered on the page.
+   */
+  componentDidMount() {
+    this.fetchData()
+    .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Fetch data
+   * @return {Promise<void>}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data: data, formData: data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  /**
+   * Set form data
+   * @param {string} formElement
+   * @param {*} value
+   */
+  setFormData(formElement, value) {
+    let formData = this.state.formData;
+    formData[formElement] = value;
+    this.setState({
+      formData: formData,
+    });
+  }
+
+  /**
+   *
+   * @return {array}
+   */
+  formattedDiagnosisEvolution() {
+    const dxEvolution = this.state.data.diagnosisEvolution;
+    let formattedDxEvolution = [];
+    dxEvolution.map((record) => {
+      const {name, project, visit, instrument, diagnosis} = record;
+      Object.entries(diagnosis).map((entry) => {
+        const [fieldName, dx] = entry;
+        formattedDxEvolution.push(
+          [
+            name,
+            project,
+            visit,
+            instrument,
+            fieldName,
+            dx,
+          ]
+        );
+      });
+    });
+    return formattedDxEvolution;
+  }
+
+  /**
+   * Render latest diagnosis element
+   * @return {JSX} - React markup for the component
+   */
+  renderLatestDiagnosis() {
+    const latestDiagnosis = this.state.data.latestDiagnosis;
+    let element = [];
+
+    latestDiagnosis.map((entry) => {
+      const projectName = this.state.data.projects[entry.ProjectID];
+      const diagnosis = entry.LatestDiagnosis ?
+        Object.values(JSON.parse(entry.LatestDiagnosis)).join(', ') : '';
+
+      element.push(
+        <StaticElement
+          key={entry.DxEvolutionID}
+          label={projectName}
```

Comment:
@jesscall how possible would it be here to render the description of the diagnosis vs the value. QPN has its diagnoses set up so that in the back-end, the options are 1,2,3,etc .. and those values are mapped to proper texts in the linst file i.e. PD, Essential Tremors, Undetermined etc. The proper text would be more useful for the front-end tab, but the value still useful to query from the DQT. I can't think of a way to do this...

Reply:
see below!